client/components/report/AttributeFilterDialog.tsx

```
@@ -0,0 +1,330 @@
+    
+    // Otherwise calculate them
+    const ranges: NumericRangeFilters = {};
+    Object.entries(availableAttributes).forEach(([attribute, values]) => {
+      if (attributeTypes[attribute] === 'numeric') {
+        const numericValues = values.map(v => v.trim() === '' ? 0 : Number(v));
+        ranges[attribute] = [
+          Math.min(...numericValues),
+          Math.max(...numericValues)
+        ];
+      }
+    });
```

Biome (1.9.4)

[error] Prefer for...of instead of forEach.

forEach may lead to performance issues when working with large arrays. When combined with functions like filter or map, this causes multiple iterations over the same type.

(lint/complexity/noForEach)